tests/test_qnode_metric_tensor.py

```diff
@@ -0,0 +1,499 @@
+
+        # diag elements
+        G1[0, 0] = np.sin(a)**2/4
+        G1[1, 1] = (
+            16 * np.cos(a) ** 2 * np.sin(b) ** 3 * np.cos(b) * np.sin(2 * c)
+            + np.cos(2 * b) * (2 - 8 * np.cos(a) ** 2 * np.sin(b) ** 2 * np.cos(2 * c))
+            + np.cos(2 * (a - b))
+            + np.cos(2 * (a + b))
+            - 2 * np.cos(2 * a)
+            + 14
+        ) / 64
+        G1[2, 2] = (3-np.cos(2*a)-2*np.cos(a)**2*np.cos(2*(b+c)))/16
+
+        assert np.allclose(G[:3, :3], G1, atol=tol, rtol=0)
+
+        # =============================================
+        # Test metric tensor of second layer is correct.
+        # We do this by computing the required expectation values
+        # numerically.
+        # The second layer includes the non_parametrized_layer,
+        # RX, RY, RZ gates (x, y, z params), a 2nd non_parametrized_layer,
+        # followed by the qml.RY(f, wires=2) operation.
+        #
+        # Observable is simply generator of:
+        #   qml.RY(f, wires=2)
+        #
+        # Note: since this layer only consists of a single parameter,
+        # only need to compute a single diagonal element.
+
+        @qnode(dev, properties={'vis_check': False})
+        def layer2_diag(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            qml.RY(f, wires=2)
+            return qml.var(qml.PauliX(1))
+
+        G2 = layer2_diag(x, y, z, h, g, f)/4
+        assert np.allclose(G[3:4, 3:4], G2, atol=tol, rtol=0)
+
+        # =============================================
+        # Test block diag metric tensor of third layer is correct.
+        # We do this by computing the required expectation values
+        # numerically using multiple circuits.
+        # The second layer includes the non_parametrized_layer,
+        # RX, RY, RZ gates (x, y, z params), and a 2nd non_parametrized_layer.
+        #
+        # Observables are the generators of:
+        #   qml.RY(f, wires=1)
+        #   qml.RZ(g, wires=2)
+        G3 = np.zeros([2, 2])
+
+        @qnode(dev, properties={'vis_check': False})
+        def layer3_diag(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            return qml.var(qml.PauliZ(2)), qml.var(qml.PauliY(1))
+
+        # calculate the diagonal terms
+        varK0, varK1 = layer3_diag(x, y, z, h, g, f)
+        G3[0, 0] = varK0/4
+        G3[1, 1] = varK1/4
+
+        assert np.allclose(G[4:6, 4:6], G3, atol=tol, rtol=0)
+
+        # ============================================
+        # Finally, double check that the entire metric
+        # tensor is as computed.
+
+        G_expected = block_diag(G1, G2, G3)
+        assert np.allclose(G, G_expected, atol=tol, rtol=0)
```

Comment:
are these just a copy and paste of the existing metric tensor tests, but using the new QNode?

Reply:
Yeah, maybe with some minor changes (like `properties={'vis_check': False}`)